File: Day75/Part5/dialogue_compression.py
def solve():
    try:
        s = input().strip()
        if not s:
            print(0)
            return
            
        freq = {}
        for c in s:
            freq[c] = freq.get(c, 0) + 1
            
        max_freq = max(freq.values())
        print(len(s) - max_freq)
            
    except EOFError:
        pass

if __name__ == "__main__":
    solve()

File: Day75/Part5/dialogue_compression.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main() {
    string s;
    if (cin >> s) {
        int n = s.length();
        if (n == 0) {
            cout << 0 << endl;
            return 0;
        }

        // Count frequency of each character
        unordered_map<char, int> freq;
        for (char c : s) {
            freq[c]++;
        }

        // Find max frequency
        int max_freq = 0;
        for (auto& p : freq) {
            max_freq = max(max_freq, p.second);
        }

        // Minimum score = length - max_freq
        cout << (n - max_freq) << endl;
    }
    return 0;
}
